### PR TITLE
Import fix

### PR DESCRIPTION
**Pull request by SteffenHeu**

### CODE REVIEW

File: mzmine-community/src/main/java/io/github/mzmine/modules/io/import_rawdata_all/AllSpectralDataImportModule.java [hunk 1]
```diff
@@ -247,6 +247,7 @@ public ExitCode runModule(final @NotNull MZmineProject project, @NotNull Paramet
 
     // after skipping already loaded
     if (checkDuplicateFilesInImportListAndProject(project, fileNames)) {
+      // in what case would this ever be triggered? ~SteffenHeu
```
Comment (robinschmid) [inline mzmine-community/src/main/java/io/github/mzmine/modules/io/import_rawdata_all/AllSpectralDataImportModule.java:250]: for duplicate files or for missing files. Why?

Comment (robinschmid) [reply mzmine-community/src/main/java/io/github/mzmine/modules/io/import_rawdata_all/AllSpectralDataImportModule.java:250]: oh ok. The first on top only searches for issues in the input list. 
This check here then double checks that the already imported files and the files to be imported do not contain duplicates. 
Like batch1/sampleA and batch2/sampleA will collide here

File: taskcontroller/src/main/java/io/github/mzmine/taskcontrol/impl/WrappedTask.java [hunk 2]
```diff
@@ -231,4 +231,8 @@ public boolean isWorkFinished() {
            (!running && getActualTask().getStatus() != TaskStatus.WAITING);
   }
 
+  @Override
+  public void setStatus(TaskStatus newStatus) {
```
Comment (robinschmid) [inline taskcontroller/src/main/java/io/github/mzmine/taskcontrol/impl/WrappedTask.java:235]: maybe do getActualTask().setStatus(newStatus)

File: mzmine-community/src/main/java/io/github/mzmine/modules/io/import_rawdata_all/AllSpectralDataImportModule.java [hunk 3]
```diff
@@ -138,12 +137,21 @@ public class AllSpectralDataImportModule implements MZmineProcessingModule {
...
+        .endsWith(".baf")) {
+      // refer to the .d folder by default
       return f.getParentFile();
+    } else if (parent.getName().endsWith(".d") && (f.getName().endsWith(".d")) && !new File(f,
+        AGILENT_ACQDATATA_FOLDER).exists()) {
+      // there also exists a .d file in the .d folder
+      return parent;
+    } else if (parent.exists() && parent.isDirectory() && Objects.requireNonNullElse(
+        parent.listFiles(p -> p != null && p.getName().startsWith("analysis.")), new File[0]).length
+        > 0 && parent.getName().endsWith(".d")) {
```
Comment (robinschmid) [inline mzmine-community/src/main/java/io/github/mzmine/modules/io/import_rawdata_all/AllSpectralDataImportModule.java:153]: The parent.getName.endsWith(.d) is a common check for all if branches. 
So maybe put it in the first one:
`if(parent==null || parent.getName().endsWith(".d"))`

This way we can skip early

File: mzmine-community/src/main/java/io/github/mzmine/modules/io/import_rawdata_all/AllSpectralDataImportModule.java [hunk 4]
```diff
@@ -226,27 +224,28 @@ public ExitCode runModule(final @NotNull MZmineProject project, @NotNull Paramet
...
     // skip files that are already loaded
-    final File[] fileNames = AllSpectralDataImportParameters.skipAlreadyLoadedFiles(project,
-        parameters);
+    final ImportFile[] filesToImport = AllSpectralDataImportParameters.skipAlreadyLoadedFiles(
+        project, parameters);
 
     // after skipping already loaded
-    if (checkDuplicateFilesInImportListAndProject(project, fileNames)) {
+    if (checkDuplicateFilesInImportListAndProject(project, filesToImport)) {
+      // in what case would this ever be triggered? ~SteffenHeu
```
Comment (robinschmid) [inline mzmine-community/src/main/java/io/github/mzmine/modules/io/import_rawdata_all/AllSpectralDataImportModule.java:248]: maybe remove this comment?
It is if files in project and import list are duplciated. So if folderA/sample1 and folderB/sample1 have the same name

File: mzmine-community/src/main/java/io/github/mzmine/modules/io/import_rawdata_all/AllSpectralDataImportModule.java [hunk 5]
```diff
@@ -260,7 +259,7 @@ public ExitCode runModule(final @NotNull MZmineProject project, @NotNull Paramet
 
     if (libraryFiles != null) {
       // no duplicate names
-      if (containsDuplicateFiles(libraryFiles, "spectral libraries in the import list")) {
+      if (containsMissingFiles(libraryFiles, "spectral library")) {
```
Comment (robinschmid) [inline mzmine-community/src/main/java/io/github/mzmine/modules/io/import_rawdata_all/AllSpectralDataImportModule.java:262]: also add a test for duplicate library files in import?
At least before we had this test.

File: mzmine-community/src/main/java/io/github/mzmine/modules/io/import_rawdata_all/AllSpectralDataImportParameters.java [hunk 6]
```diff
@@ -128,29 +135,37 @@ public static List<RawDataFile> getLoadedRawDataFiles(MZmineProject project,
...
    */
-  public static File[] skipAlreadyLoadedFiles(MZmineProject project,
+  public static ImportFile[] skipAlreadyLoadedFiles(MZmineProject project,
       final ParameterSet parameters) {
-    var loadedFiles = getLoadedRawDataFiles(project, parameters);
-    var loadedAbsFiles = loadedFiles.stream().map(RawDataFile::getAbsoluteFilePath)
-        .collect(Collectors.toSet());
+    // all files that should be loaded
+    // need to validate bruker paths and use absolute file paths as they are used in RawDataFile
+    Set<ImportFile> filesToLoad = streamValidatedFiles(parameters).collect(Collectors.toSet());
```
Comment (robinschmid) [inline mzmine-community/src/main/java/io/github/mzmine/modules/io/import_rawdata_all/AllSpectralDataImportParameters.java:142]: remove this intermediate set - it is unused.
This stream can be inlined into the return statement.

`return streamValidatedFiles(parameters).filter ....`

**Review by robinschmid — CHANGES_REQUESTED** [review]
Looks good to me only a few minor things
